### PR TITLE
prompt git branch

### PR DESCRIPTION
**Pull request by laerus**
#1778

### CODE REVIEW

**Comment by laerus** [comment]
nope, porcelain is strange.

File: xonsh/prompt/vc_branch.py [hunk 1]
```diff
@@ -16,16 +16,13 @@
...
                                          stderr=subprocess.DEVNULL)
     except (subprocess.CalledProcessError, OSError):
         q.put(None)
     else:
-        status = status.decode().split()
-        if status[2] == 'at':
-            q.put(status[3])
-        else:
-            q.put(status[2])
+        status = status.decode().splitlines()[0].split()[1]
```
Comment (scopatz) [inline xonsh/prompt/vc_branch.py]: Calls to decode should use the `XONSH_ENCODING` and `XONSH_ENCODING_ERRORS` env vars

**Comment by scopatz** [comment]
Thanks @laerus!

File: xonsh/prompt/vc_branch.py (hunk 1 above)
Comment (laerus) [inline xonsh/prompt/vc_branch.py]: shouldn't `tools.decode` respect these envvars too?

Comment (laerus) [inline xonsh/prompt/vc_branch.py]: ok it seems that `tools.decode_bytes` does this, but the docs are little misleading.

Comment (scopatz) [inline xonsh/prompt/vc_branch.py]: Ahh yeah! Thanks for fixing this up.

**Comment by scopatz** [comment]
Thanks @laerus!

**Pull request merged**